apps/src/templates/census2017/YourSchool.jsx

```diff
@@ -27,25 +30,58 @@ const styles = {
   }
 };
 
-export default class YourSchool extends React.Component {
+export default class YourSchool extends Component {
   static propTypes = {
-    alertHeading: React.PropTypes.string,
-    alertText: React.PropTypes.string,
-    alertUrl: React.PropTypes.string,
-    hideMap: React.PropTypes.bool
+    alertHeading: PropTypes.string,
+    alertText: PropTypes.string,
+    alertUrl: PropTypes.string,
+    hideMap: PropTypes.bool
   };
 
+  constructor(props) {
+    super(props);
+    this.responsive = new Responsive();
+    this.state = {
+      windowWidth: $(window).width(),
+      windowHeight: $(window).height(),
+      mobileLayout: this.responsive.isResponsiveCategoryInactive('md')
+    };
+  }
+
   componentDidMount() {
     if (!this.props.hideMap) {
       $('#map').appendTo(ReactDOM.findDOMNode(this.refs.map)).show();
     }
+    // Resize handler.
+    window.addEventListener('resize', _.debounce(this.onResize, 100).bind(this));
+  }
+
+  onResize() {
+    const windowWidth = $(window).width();
+    const windowHeight = $(window).height();
+
+    // We fire window resize events when the grippy is dragged so that non-React
+    // controlled components are able to rerender the editor. If width/height
+    // didn't change, we don't need to do anything else here
+    if (windowWidth === this.state.windowWidth &&
+        windowHeight === this.state.windowHeight) {
+      return;
+    }
+
+    this.setState({
+      windowWidth: $(window).width(),
+      windowHeight: $(window).height()
+    });
+
+    this.setState({mobileLayout: this.responsive.isResponsiveCategoryInactive('md')});
```

Comment:
So it turns out we're not really using any of this state.  However, changing any state at all has the effect or triggering a re-rendering, which we do want.  (See "By default, when your component's state or props change, your component will re-render." at https://facebook.github.io/react/docs/react-component.html)  However, I think a better technique will be to force an update.  (See https://facebook.github.io/react/docs/react-component.html#forceupdate)

Reply:
What about the part where the doc also says "Normally you should try to avoid all uses of forceUpdate()" ?

Reply:
Good question.  I guess the right React solution is to set state, and then have that changed state be meaningful.  (At the moment, we don't use this state for anything in either /courses or here, and the Responsive component just re-retrieves the window width when it's involved in the re-render).  I think this means having the new state (i.e. the new window dimensions) passed into the Responsive component which will then use them.